src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -125,30 +135,82 @@ public SharingSettingDeleteResponse delete(UUID consortiumId, UUID settingId, Sh
     log.info("start:: The Sharing Settings for settingId '{}' and '{}' unique tenant(s) were successfully deleted from the database", sharingSettingRequest.getSettingId(), publicationDeleteRequest.getTenants().size());
 
     // we create PC request with POST and PUT Http method to create settings as a consortia-system-user
+    UUID pcId;
+    SharingSettingDeleteResponse sharingSettingDeleteResponse;
     try (var ignored = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(folioExecutionContext.getTenantId()))) {
-      UUID pcId = publishRequest(consortiumId, publicationDeleteRequest);
-      return new SharingSettingDeleteResponse()
+      pcId = publishRequest(consortiumId, publicationDeleteRequest);
+      sharingSettingDeleteResponse = new SharingSettingDeleteResponse()
         .pcId(pcId);
     }
+    asyncTaskExecutor.execute(() -> updateSettingsForFailedTenants(consortiumId, pcId, sharingSettingRequest));
+    return sharingSettingDeleteResponse;
+  }
+
+  private UUID publishRequest(UUID consortiumId, PublicationRequest publicationRequest) {
+    if (CollectionUtils.isNotEmpty(publicationRequest.getTenants())) {
+      return publicationService.publishRequest(consortiumId, publicationRequest).getId();
+    }
+    log.info("publishRequest:: Tenant list of publishing for http method: {} is empty", publicationRequest.getMethod());
+    return null;
+  }
+
+  private void updateSettingsForFailedTenants(UUID consortiumId, UUID publicationId, SharingSettingRequest sharingSettingRequest) {
+    log.debug("updateSettingsForFailedTenants:: Trying to update settings for failed tenants for consortiumId={} publicationId={} and sharingSettingRequestId={}",
+      consortiumId, publicationId, sharingSettingRequest.getSettingId());
+    boolean desiredResultReceived = false;
+    long startTime = System.currentTimeMillis();
+    long timeout = 3000;
+
+    while (Boolean.FALSE.equals(desiredResultReceived) && System.currentTimeMillis() - startTime < timeout) {
+      boolean isPublicationStatusExists = publicationService.checkPublicationStatusExists(publicationId);
```

Comment:
If status is not exists - we can de-doss DB for 3 seconds, thinking how to improve